### PR TITLE
Fix in-place to views condition

### PR DESCRIPTION
**Pull request by crcrpar**
## What does this PR do?

Special case batch norm and avg/max-pool in the checker as they would not return a view as their output.

Add a test that make sure that thunder can execute `torchvision.models.resnet18`

### CODE REVIEW

File: thunder/tests/test_inplace_functionalization.py [hunk 1]
```diff
@@ -147,3 +148,22 @@ def f_with_contiguous(a: torch.Tensor) -> torch.Tensor:
...
+
+
+@requiresCUDA
+def test_parse_resnet18():
+    import thunder
+
+    torchvision = pytest.importorskip("torchvision")
+
+    device = torch.device("cuda")
+    dtype = torch.bfloat16 if device_supports_bf16(device) else torch.float16
```
Comment (t-vi) [inline thunder/tests/test_inplace_functionalization.py:160]:
```suggestion
    dtype = torch.bfloat32
```

File: thunder/tests/test_inplace_functionalization.py [hunk 2]
```diff
@@ -147,3 +148,22 @@ def f_with_contiguous(a: torch.Tensor) -> torch.Tensor:
...
+    model: nn.Module = torchvision.models.resnet18().to(device=device, dtype=dtype)
+    ref_model: nn.Module = torchvision.models.resnet18().to(device=device, dtype=dtype)
+    ref_model.load_state_dict(model.state_dict())
+
+    jitted = thunder.jit(model)
+    x = torch.randn((1, 3, 224, 224), device=device, dtype=dtype)
+    # FIXME(crcrpar): Why...?
+    with pytest.raises(AssertionError, match="Tensor-likes are not close!"):
```
Comment (t-vi) [inline thunder/tests/test_inplace_functionalization.py:168]:
```suggestion
```

File: thunder/tests/test_inplace_functionalization.py [hunk 3]
```diff
@@ -147,3 +148,22 @@ def f_with_contiguous(a: torch.Tensor) -> torch.Tensor:
...
+        torch.testing.assert_close(jitted(x), ref_model(x))
```
Comment (t-vi) [inline thunder/tests/test_inplace_functionalization.py:169]:
```suggestion
      torch.testing.assert_close(jitted(x), ref_model(x), atol=1e-3, rtol=1e-3)
```

File: thunder/tests/test_inplace_functionalization.py [hunk 4]
```diff
@@ -147,3 +148,22 @@ def f_with_contiguous(a: torch.Tensor) -> torch.Tensor:
...
+
+    device = torch.device("cuda")
+    dtype = torch.bfloat16 if device_supports_bf16(device) else torch.float16
+    model: nn.Module = torchvision.models.resnet18().to(device=device, dtype=dtype)
+    ref_model: nn.Module = torchvision.models.resnet18().to(device=device, dtype=dtype)
+    ref_model.load_state_dict(model.state_dict())
+
+    jitted = thunder.jit(model)
+    x = torch.randn((1, 3, 224, 224), device=device, dtype=dtype)
+    # FIXME(crcrpar): Why...?
```
Comment (t-vi) [inline thunder/tests/test_inplace_functionalization.py:167]:
```suggestion
    # FIXME(crcrpar): Investigate numerical accuracy.
```

**Review by t-vi — APPROVED** [review]
Thank you @crcrpar 

Supergreat milestone running stock ResNet18! :rocket: 
I think it would be better to test fp32 + generous bounds, but in addition we should file an issue about the accuracy. WDYT?

**Comment by t-vi** [comment]
I looked at the numerical accuracy a bit more and does not seem to happen with eval mode, so maybe something with the training mode batch norm.
